### DISCUSSION TITLE
Hybridized methods

### DISCUSSION THREAD

**Discussion by lindsayad** [mfem/mfem#3830] [General]
I was peeking around today and saw that @tzanio added the Hybridization class in 2016 (the formulas in the doxygen for that class don’t appear to be rendering on the HTML pages FWIW). I’ve been reading a lot over the last couple days about hybridizable methods and a reviewer recently suggested to me that I look into them for CFD. 

I’m curious what the MFEM community’s experience with hybrid methods has been? Is it oft used? I believe the method requires local solves in order to eliminate the trace degrees of freedom. How expensive are these? Worth the reduction in size of the global system?

This is a situation in which it would be nice if MOOSE was based on top of MFEM; a lot of nice features to play with! I am continually exploring best practices for solving realistic flow problems in advanced nuclear reactor designs; these generally involve high Reynolds’s numbers, high aspect ratio elements at boundaries, and potential momentum sources/sinks (one example being solidification in a molten salt reactor accident scenario). The NEAMS program is currently focused on finite volumes but my belief is that we can do better with finite elements (although I could be wrong!).

**Comment by tzanio** [comment]
ping: @thorvath12

**Comment by thorvath12** [comment]
Hi @lindsayad 

I have used hybridization for different problems (incompressible Navier-Stokes, MHD). I created a separate branch (hdg) and have been working on that this summer. I will push some significant changes to that in the coming weeks and give a talk at the MFEM Community Workshop in October.

The reduction in the DoFs can be significant, depending on the polynomial order used. Whether it is cheap or expensive is a good question. Preconditioning the system and solving it in a meaningful way might be challenging. But some people are working on those problems.

**Reply by lindsayad** [reply]
Thank you for the reply @thorvath12. Some other MOOSE team members and I will be attending the community workshop, so I look forward to your talk! I submitted a research proposal at Idaho National Lab a couple weeks ago to explore HDG for MOOSE so I am very much interested in learning more and reusing what is already known and developed within the open source community. Navier-Stokes is my personal favorite application and my proposal discussed the combination of HDG with current research in block preconditioning for NS, e.g. augmented Lagrange, least squares commutator, etc.
